### PR TITLE
readme, sql: add brief introduction about optimizer

### PR DESCRIPTION
**Pull request by zz-jason**

### CODE REVIEW

**Comment by zz-jason** [comment]
@shenli @lilin90 PTAL

File: sql/sql-optimizer-overview.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---
+title: SQL 优化流程简介
+category: user guide
+---
+
+# TiDB SQL 优化器概览
```
Comment (lilin90) [inline sql/sql-optimizer-overview.md]: It'd be better to keep this consistent with the `title` name in metadata.

File: sql/sql-optimizer-overview.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+在 TiDB 中，SQL 优化过程分为逻辑优化和物理优化两个阶段。
+
+## 逻辑优化简介
+
+逻辑优化是基于规则的优化，对输入的逻辑执行计划按顺序应用一些使得整个逻辑执行计划变的更好的优化规则，这些优化规则包括：
```
Comment (lilin90) [inline sql/sql-optimizer-overview.md]: It seems that this sentence is a little bit too long. Please update it to:

逻辑优化是基于规则的优化，对输入的逻辑执行计划按顺序应用一些**优化规则，从而使整个逻辑执行计划变得更好。这些优化规则包括：**

**Comment by lilin90** [comment]
@shenli PTAL. Thanks!

**Comment by zz-jason** [comment]
@lamxTyler @eurekaka PTAL

**Comment by shenli** [comment]
LGTM

**Pull request merged**